packages/core/src/connectivity/scp-cf/destination/destination.ts

```diff
@@ -55,7 +55,40 @@ export function parseDestination(
     }
   );
 
-  return sanitizeDestination(destination);
+  const additionalHeadersAndQueryParameters =
+    getAdditionalHeadersAndQueryParameters(destinationConfig);
+
+  return sanitizeDestination({
+    ...destination,
+    ...additionalHeadersAndQueryParameters
+  });
+}
+
+function getAdditionalHeadersAndQueryParameters(
+  destinationConfig: DestinationConfiguration
+): Record<string, any> {
+  const configDestinationPropertyMapping = {
+    'URL.headers.': 'headers',
+    'URL.queries.': 'queries'
+  };
+
+  return Object.entries(destinationConfig).reduce(
+    (additionalHeaders: Record<string, any>, [originalKey, value]) => {
+      const mapping = Object.entries(configDestinationPropertyMapping).find(
+        ([configProperty]) => originalKey.startsWith(configProperty)
+      );
+      if (mapping) {
+        const [configProperty, destinationProperty] = mapping;
+        if (!additionalHeaders[destinationProperty]) {
+          additionalHeaders[destinationProperty] = {};
+        }
+        const headerKey = originalKey.replace(configProperty, '');
+        additionalHeaders[destinationProperty][headerKey] = value;
+      }
+      return additionalHeaders;
```

Comment:
Are the query parameters also handled here? In general I do not like this method because it is hard to read. My suggestion:
- One method: getDestinationPropertiesMathcingPattern(): {key:value}.
- One method: adjustKeys() replacing the keys